sdk/lib/rtl/path.c

```diff
@@ -446,20 +446,144 @@ NTAPI
 RtlpApplyLengthFunction(IN ULONG Flags,
                         IN ULONG Type,
                         IN PVOID UnicodeStringOrUnicodeStringBuffer,
-                        IN PVOID LengthFunction)
+                        IN NTSTATUS(NTAPI* LengthFunction)(ULONG, PUNICODE_STRING, PULONG))
 {
-    UNIMPLEMENTED;
-    return STATUS_NOT_IMPLEMENTED;
+    NTSTATUS Status;
+    PUNICODE_STRING String;
+    ULONG Length;
+
+    if (Flags)
+    {
+        DPRINT1("ERROR: unhandled Flags = 0x%x\n", Flags);
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    if (UnicodeStringOrUnicodeStringBuffer == NULL)
+    {
+        DPRINT1("ERROR: UnicodeStringOrUnicodeStringBuffer == NULL\n");
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    if (Type == sizeof(UNICODE_STRING))
+    {
+        String = (PUNICODE_STRING)UnicodeStringOrUnicodeStringBuffer;
+    }
+    else if (Type == sizeof(RTL_UNICODE_STRING_BUFFER))
+    {
+        String = &((PRTL_UNICODE_STRING_BUFFER)UnicodeStringOrUnicodeStringBuffer)->String;
+    }
+    else
+    {
+        DPRINT1("ERROR: Type = %u\n", Type);
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    Length = 0;
+    Status = LengthFunction(0, String, &Length);
+    if (!NT_SUCCESS(Status))
+        return Status;
+
+    if (Length > UNICODE_STRING_MAX_CHARS)
+        return STATUS_NAME_TOO_LONG;
+
+    String->Length = (USHORT)(Length * sizeof(WCHAR));
+
+    if (Type == sizeof(RTL_UNICODE_STRING_BUFFER))
+    {
+        String->Buffer[String->Length / 2] = UNICODE_NULL;
+    }
+
+    return STATUS_SUCCESS;
 }
 
 NTSTATUS
 NTAPI
 RtlGetLengthWithoutLastFullDosOrNtPathElement(IN ULONG Flags,
```

Comment:
What about mentioning (in the commit log) that a documentation of this (undocumented) function can be found at:
http://undoc.airesoft.co.uk/ntdll.dll/RtlGetLengthWithoutLastFullDosOrNtPathElement.php ?

Reply:
It's mentioned in the apitest.

Reply:
Yes it's basically hidden inside. That's why I didn't see it.

Reply:
We don't generally litter our source tree with comments to documents, unless there is something special there.

Reply:
not the source tree. The commit log.